### PR TITLE
Add tests for categorical with null ea as input

### PR DESCRIPTION
**Pull request by phofl**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

cc @jreback Is this what you had in mind for ExtensionArrays and Categoricals?

### CODE REVIEW

File: pandas/tests/arrays/categorical/test_constructors.py [hunk 1]
```diff
@@ -682,3 +682,11 @@ def test_interval(self):
         expected_codes = np.array([0, 1], dtype="int8")
         tm.assert_numpy_array_equal(cat.codes, expected_codes)
         tm.assert_index_equal(cat.categories, idx)
+
+    @pytest.mark.parametrize("nullable", [pd.NA, NaT])
```
Comment (jreback) [inline pandas/tests/arrays/categorical/test_constructors.py:686]: actually you can use the null_fixture here

**Review by jreback — CHANGES_REQUESTED** [review]
yes this is what i was looking

File: pandas/tests/arrays/categorical/test_constructors.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/arrays/categorical/test_constructors.py]: Thx, done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@phofl can you merge master and ping on green.

**Pull request merged**